baseclasses/utils.py

```diff
@@ -91,6 +103,11 @@ def __new__(self, *args, **kwargs):
         self.map = {}
         return super().__new__(self)
 
+    def __eq__(self, other):
```

Comment:
I am not really sure about the syntax and the role of this method

Reply:
This function is used for comparisons, and is called when you run `obj1 == obj2`.